src/linux/system.rs

```diff
@@ -697,6 +697,21 @@ impl SystemExt for System {
     fn os_version(&self) -> Option<String> {
         get_system_info_android(InfoType::OsVersion)
     }
+
+    #[cfg(not(target_os = "android"))]
+    fn distribution_id(&self) -> String {
+        get_system_info_linux(
+            InfoType::DistributionID,
+            Path::new("/etc/os-release"),
+            Path::new(""),
+        )
+        .unwrap_or(std::env::consts::OS.to_owned())
```

Comment:
To avoid allocating every time:

```suggestion
        .unwrap_or_else(|| std::env::consts::OS.to_owned())
```

Reply:
Yea, like I said I was working on test failures already.